pkg/fakerp/aad.go

```diff
@@ -73,8 +74,17 @@ func (am *aadManager) ensureApp(ctx context.Context, displayName string, p *api.
 	}
 	p.ClientID = *app.AppID
 
-	sp, err := am.sc.Create(ctx, graphrbac.ServicePrincipalCreateParameters{
-		AppID: app.AppID,
+	var sp graphrbac.ServicePrincipal
+	err = wait.PollInfinite(5*time.Second, func() (bool, error) {
+		sp, err = am.sc.Create(ctx, graphrbac.ServicePrincipalCreateParameters{
+			AppID: app.AppID,
+		})
+		// ugh: Azure client library doesn't have the types registered to
+		// unmarshal all the way down to this error code natively :-(
+		if strings.Contains(err.Error(), "NoBackingApplicationObject") {
+			return false, nil
+		}
+		return err == nil, err
 	})
 	if err != nil {
```

Comment:
Isn't `err` here always going to be a tuple (bool, err) and hence will never be nil?

Reply:
No, err is of type error throughout here.  Not quite sure what's driving that thought?

Reply:
My misreading of the code since the return statements above all return tuples :) I went and read through the [`wait` documentation](https://godoc.org/k8s.io/apimachinery/pkg/util/wait#PollInfinite) and understand better what's going on here now.